### PR TITLE
Move (almost all) reference apps to separate repo

### PR DESCRIPTION
**Pull request by jan-warchol**
* Remove (almost all) reference apps

  They have their own repo, https://github.com/scramjetorg/reference-apps
  However, a few of them are quite tightly coupled to BDD tests code, so
  for now I'm keeping them - they should be decoupled later.

* Add script for downloading prebuilt refapps (for local testing)

* CI: replace building refapps with downloading prebuilt ones

  We now have a separate repo for reference apps, with tar.gz files
  available as released artifacts. This is much faster than building them
  every time.

Basically, most of this PR is just deleting files. Fortunately GH folds their previews, so you can simply focus on reviewing visible code changes.

### CODE REVIEW

**Review by Tatarinho — APPROVED** [review]

File: bdd/step-definitions/e2e/host-steps.ts [hunk 1]
```diff
@@ -221,10 +221,16 @@ When("wait for {string} ms", async (timeoutMs: number) => {
 });
 
 When("sequence {string} loaded", { timeout: 50000 }, async function(this: CustomWorld, packagePath: string) {
```
Comment (Tatarinho) [inline bdd/step-definitions/e2e/host-steps.ts:223]: two similar steps, maybe it is good idea to remove it?

Comment (jan-warchol) [reply bdd/step-definitions/e2e/host-steps.ts:223]: To be honest, I really would like not deal with them in this PR. I've tried untangling step definitions in the past and it usually turned out that they are interconnected in some weird ways - changing one requires changing some global variables, which affect other steps, and so on...

So, while it seems trivial, I'd say there's a 40% chance that it will eat up half a day.

**Review by a-tylenda — APPROVED** [review]

File: bdd/step-definitions/e2e/host-steps.ts (hunk 1 above)
Comment (a-tylenda) [reply bdd/step-definitions/e2e/host-steps.ts:223]: I agree with @jan-warchol , I also tried once to unify similar steps and failed, but sooner or later it will have to be done I guess.
